### ISSUE TITLE
jsx-use-translation-function overlooks ternary expressions

### ISSUE THREAD

**Issue by karol-majewski** [palantir/tslint-react#155]
Hello and thank you for your work. I believe the `jsx-use-translation-function` rule should look into what an expression evaluates to, not only for string literals alone.

#### Linted code

```tsx
<div>{true ? 'Told ya' : 'Not really'}</div>
```

#### Expected result

```
String literals are disallowed as JSX. Use a translation function (jsx-use-translation-function)
```

#### Actual result

No error.

#### Configuration

```json
{
  "extends": [
    "tslint-react"
  ],
  "rules": {
    "jsx-use-translation-function": true
  }
}
```

- `tslint`: `5.9.1`
- `tslint-react`: `3.5.1`

**Comment by adidahiya** [comment]
Static code analysis tools like TSLint avoid evaluating code for good reason: it is expensive and error-prone. I think this is out of scope.
